src/main/java/io/temporal/internal/replay/ReplayWorkflowExecutor.java

```diff
@@ -622,27 +341,76 @@ public void close() {
 
   @Override
   public Optional<Payloads> handleQueryWorkflowTask(
-      PollWorkflowTaskQueueResponseOrBuilder response, WorkflowQuery query) throws Throwable {
+      PollWorkflowTaskQueueResponseOrBuilder workflowTask, WorkflowQuery query) {
     lock.lock();
     try {
       AtomicReference<Optional<Payloads>> result = new AtomicReference<>();
-      handleWorkflowTaskImpl(response, () -> result.set(workflow.query(query)));
+      handleWorkflowTaskImpl(workflowTask);
+      result.set(workflow.query(query));
       return result.get();
     } finally {
       lock.unlock();
     }
   }
 
-  public Consumer<HistoryEvent> getLocalActivityCompletionSink() {
-    return localActivityCompletionSink;
+  private void processLocalActivityRequests(long startTime) {
+    Duration maxProcessingTime = getWorkflowTaskTimeout().multipliedBy(4).dividedBy(5);
+    while (true) {
+      List<ExecuteLocalActivityParameters> laRequests =
+          workflowStateMachines.takeLocalActivityRequests();
+      long timeoutInterval = (long) ((System.currentTimeMillis() - startTime) * 0.5);
```

Comment:
this is a strange name for a variable (timeoutInterval) especially as it just seems to represent the elapsed amount of time?

Reply:
Refactored this part of the code.